modules/network/src/integrationTest/java/org/apache/ignite/internal/network/netty/ItConnectionManagerTest.java

```diff
@@ -295,6 +299,99 @@ public void testStopTwice() throws Exception {
         server.close();
     }
 
+    /**
+     * Tests that if two nodes are opening channels to each other, only one channel survives.
+     *
+     * @throws Exception If failed.
+     */
+    @RepeatedTest(100)
+    public void testOneChannelLeftIfConnectToEachOther() throws Exception {
+        try (
+                ConnectionManagerWrapper manager1 = startManager(4000);
+                ConnectionManagerWrapper manager2 = startManager(4001)
+        ) {
+            CompletableFuture<NettySender> fut1 = manager1.openChannelTo(manager2).toCompletableFuture();
+            CompletableFuture<NettySender> fut2 = manager2.openChannelTo(manager1).toCompletableFuture();
+
+            NettySender sender1 = null;
+            NettySender sender2 = null;
+
+            try {
+                sender1 = fut1.join();
+            } catch (Exception ignored) {
+                // No-op.
```

Comment:
Is it correct to suggest that for exactly one `.join()` a `ChannelAlreadyExistsException` will be thrown? If yes, let's assert that the exception is what we expect it to be